src/main.rs

```diff
@@ -459,31 +525,38 @@ impl<'a> ResolvedPackage<'a> {
                     cwd
                 )
             )?;
-            writeln!(f, "features = builtins.concatLists [")?;
-            for (feature, optionality) in self.features.iter() {
-                let mut f = f.indent(2);
-                match optionality
-                    .to_expr(outer.root_features, n_root_pkgs)
-                    .simplify()
-                {
-                    True => writeln!(f, "[ {:?} ]", feature)?,
-                    expr => writeln!(f, "({} ({}) {:?})", outer.optional, expr.to_nix(), feature)?,
+            if self.features.len() != 0 {
+                write!(f, "features = [")?;
+                for (feature, optionality) in self.features.iter() {
+                    let mut f = f.indent(2);
+                    match optionality
+                        .to_expr(outer.root_features, n_root_pkgs)
+                        .simplify()
+                    {
+                        True => write!(f, " {:?}", feature)?,
+                        expr => {
+                            write!(f, " ({} ({}) {:?})", outer.optional, expr.to_nix(), feature)?
+                        }
+                    }
                 }
+                writeln!(f, " ];")?;
             }
-            writeln!(f, "];")?;
 
             for (attr, kind) in &[
                 ("dependencies", DependencyKind::Normal),
                 ("devDependencies", DependencyKind::Development),
                 ("buildDependencies", DependencyKind::Build),
             ] {
-                writeln!(f, "{} = {{", attr)?;
-
-                for ((dep_id, _), rdep) in self
+                let deps: Vec<_> = self
                     .deps
                     .iter()
                     .filter(|((_, dep_kind), _)| dep_kind == kind)
-                {
+                    .collect();
```

Comment:
Not a request change, but you can avoid materializing the iterator by using `iter.peekable().peek().is_none()`.

Reply:
Nice. I was looking for something like that... the `.collect()` seemed forced. Thanks.